eZ/Publish/Core/Persistence/Legacy/Content/FieldValue/Converter/RelationConverter.php

```diff
@@ -66,30 +67,103 @@ public function toFieldValue(StorageFieldValue $value, FieldValue $fieldValue)
      */
     public function toStorageFieldDefinition(FieldDefinition $fieldDef, StorageFieldDefinition $storageDef)
     {
-        // Selection method, 0 = browse, 1 = dropdown
-        $storageDef->dataInt1 = (int)$fieldDef->fieldTypeConstraints->fieldSettings['selectionMethod'];
+        $fieldSettings = $fieldDef->fieldTypeConstraints->fieldSettings;
+        $doc = new DOMDocument('1.0', 'utf-8');
+        $root = $doc->createElement('related-objects');
+        $doc->appendChild($root);
 
-        // Selection root, location ID
-        $storageDef->dataInt2 = (int)$fieldDef->fieldTypeConstraints->fieldSettings['selectionRoot'];
+        $constraints = $doc->createElement('constraints');
+        if (!empty($fieldSettings['selectionContentTypes'])) {
+            foreach ($fieldSettings['selectionContentTypes'] as $typeIdentifier) {
```

Comment:
The mapping of settings to XML, and vice-versa below, is a bit long for procedural code within that method.

It would be more readable if moved to a private class method. The option above is to move it to a dedicated service/class. Doing so would ease unit testing.